src/Microsoft.AspNetCore.Razor.Evolution/DefaultRazorIRLoweringPhase.cs

```diff
@@ -206,6 +206,44 @@ public override void VisitImportSpan(AddImportChunkGenerator chunkGenerator, Spa
                 Namespace.Children.Insert(i, @using);
             }
 
+            public override void VisitDirectiveToken(DirectiveTokenChunkGenerator chunkGenerator, Span span)
+            {
+                Builder.Add(new DirectiveTokenIRNode()
+                {
+                    Content = span.Content,
+                    Descriptor = chunkGenerator.Descriptor,
+                    SourceLocation = span.Start,
+                });
+            }
+
+            public override void VisitStartDirectiveBlock(DirectiveChunkGenerator chunkGenerator, Block block)
+            {
+                Builder.Push(new DirectiveIRNode()
+                {
+                    Name = chunkGenerator.Descriptor.Name,
+                    Descriptor = chunkGenerator.Descriptor,
+                });
+            }
+
+            public override void VisitEndDirectiveBlock(DirectiveChunkGenerator chunkGenerator, Block block)
+            {
+                var directiveNode = (DirectiveIRNode)Builder.Pop();
+
+                var tokens = directiveNode.Children.OfType<DirectiveTokenIRNode>().ToList();
+                var exceptTokens = directiveNode.Children.Except(tokens).ToList();
```

Comment:
I get **why** you're doing this, but wouldn't it be better to just leave the `Children` as is and make `Tokens` do something like:

```
public IEnumerable<DirectiveTokenIRNode> Tokens => Children.OfType<DirectiveTokenIRNode>();
```